src/main/java/task/Task.java

```diff
@@ -0,0 +1,179 @@
+    private LocalDateTime dateAdded;
+
+    /**
+     * Calls the constructor to create a Task.
+     *
+     * @param taskName The name of the task.
+     * @return Returns a Task object.
+     */
+    public static Task of(String taskName) {
+        return new ToDos(taskName);
+    }
+
+    /**
+     * Calls the constructor to create a Task.
+     *
+     * @param taskName The name of the task.
+     * @param dayDate The date and time of the deadline of the task.
+     * @return Returns a Task object.
+     */
+    public static Task of(String taskName, LocalDateTime dayDate) {
+        return new Deadlines(taskName, dayDate);
+    }
+
+    /**
+     * Calls the constructor to create a Task.
+     *
+     * @param taskName The name of the task.
+     * @param startDayDateTime The date and time of the start of the event.
+     * @param endDayDateTime The date and time of the end of the event.
+     * @return Returns a Task object.
+     */
+    public static Task of(String taskName, LocalDateTime startDayDateTime, LocalDateTime endDayDateTime) {
+        return new Event(taskName, startDayDateTime, endDayDateTime);
+    }
+
+    private Task(String taskName) {
+        this.taskName = taskName;
+        this.isDone = false;
+        this.dateAdded = LocalDateTime.now();
+    }
+
+    /**
+     * The string that represents the object.
+     *
+     * @return Returns a string that represents the object.
+     */
+    @Override
+    public String toString() {
+        return ("[" + (this.isDone ? "X] " : " ] ") + this.taskName);
+    }
+
+    /**
+     * Mark the task as done.
+     */
+    public void setDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Mark the task as undone.
+     */
+    public void setNotDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Check if the task is done.
+     *
+     * @return Returns true if the task is done, false if otherwise.
+     */
+    public boolean getIsDone() {
+        return this.isDone;
+    }
+
+    /**
+     * Returns the name of the task.
+     *
+     * @return Returns the name of the task.
+     */
+    public String getTaskName() {
+        return this.taskName;
+    }
+
+    /**
+     * Writes the initial command the user put that resulted in the task.
+     *
+     * @return Returns the initial command the user put for this task.
+     */
+    public String write() {
+        return "task " + this.taskName + "\n";
+    }
+
+
+    private static class ToDos extends Task {
+        private ToDos(String taskName) {
+            super(taskName);
+        }
+
+        /**
+         * @inheritDoc
+         */
+        @Override
+        public String toString() {
+            return "[T]" + super.toString();
+        }
+
+        /**
+         * @inheritDoc
+         */
+        @Override
+        public String write() {
+            return "todo " + super.taskName + "\n";
+        }
+    }
+
+    private static class Deadlines extends Task {

```

Comment:
Perhaps extracting these private nested classes as standalone public classes would be better?